src/codeu/chat/server/Server.java

```diff
@@ -196,6 +232,22 @@ public void run() {
             onBundle(bundle);
             lastSeen = bundle.id();
           }
+          
+          try{
+        	  persistentDataWriter = new PrintWriter(new FileWriter(persistentFile, true));
+          
+        	  while(!persistentLog.isEmpty()){
+        		  persistentDataWriter.println(persistentLog.pop());
+        	  }
+          }  
+          catch(Exception e){
+        	  LOG.error(e, "Unable to read from file");

```

Comment:
This is actually write and not read, since it's not getting write permissions that would get you to this catch.